saja/Task.py

```diff
@@ -0,0 +1,100 @@
+import uuid
+from enum import Enum
+from datetime import datetime
+
+class Status(Enum):
+    """Enum to represent the status of a Task."""
+    InProgress = "InProgress"
+    Completed = "Completed"
+
+class Priority(Enum):
+    """Enum to represent the priority of a Task."""
+    Low = "Low"
+    Medium = "Medium"
+    High = "High"
+
+class Task:
+    """Class to represent a Task with a title, description, due date, and status."""
+
+    def __init__(self, title, description, due_date, status: Status):
+        """
+        Initialize a Task instance.
+
+        Parameters:
+        - title: The title of the task.
+        - description: A brief description of the task.
+        - due_date: The due date of the task in the format 'YYYY-MM-DD'.
+        - status: The current status of the task, should be an instance of Status enum.
+        """
+        self.id = uuid.uuid4()  # Generate a unique identifier for the task.
```

Comment:
Good choice, using the UUID is the best option especially when we have multiple replicas of the same app.